MigrationTutorial/Blog-Post.md

```diff
@@ -0,0 +1,658 @@
+
+    [Required]
+    private string _Type { get; set; } = string.Empty;
+
+    public Consumable(string productId = "")
+    {
+        ProductId = productId;
+    }
+
+    private Consumable() { }
+}
+```
+---
+
+#### *(8) Model V2.Department*
+```cs
+public class Department : RealmObject
+{
+    [PrimaryKey]
+    public ObjectId Id { get; private set; } = ObjectId.GenerateNewId();
+
+    [Required]
+    public string Name { get; set; }
+
+    public Employee Head { get; set; }
+
+    [Backlink(nameof(Employee.Department))]
+    public IQueryable<Employee> Employees { get; }
+}
+```
+
+---
+
+#### *(9) Model V2.Customer*
+```cs
+public class Customer : RealmObject
+{
+    [PrimaryKey]
+    public ObjectId Id { get; private set; } = ObjectId.GenerateNewId();
+
+    [Required]
+    public string Name { get; set; }
+
+    public string Location { get; set; }
+}
+```
+
+---
+
+#### *(10) Model V2.Supplier*
+```cs
+public class Supplier : RealmObject
+{
+    [PrimaryKey]
+    public ObjectId Id { get; private set; } = ObjectId.GenerateNewId();
+
+    public string Name { get; set; }
+
+    private ISet<string> _SuppliedTypes { get; }
+
+    [Ignored]
+    public ISet<ConsumableType> SuppliedTypes { get; } = new HashSet<ConsumableType>();
+
+    public void AddConsumableType(ConsumableType[] consumables)
+    {
+        foreach (var consumable in consumables)
+        {
+            _SuppliedTypes.Add(consumable.ToString());
+            SuppliedTypes.Add(consumable);
+        }
+    }
+
+    public void AddConsumableType(ConsumableType consumable)
+    {
+        _SuppliedTypes.Add(consumable.ToString());
+        SuppliedTypes.Add(consumable);
+    }
+
+    public void RemoveConsumableType(ConsumableType[] consumables)
+    {
+        foreach (var consumable in consumables)
+        {
+            _SuppliedTypes.Remove(consumable.ToString());
+            SuppliedTypes.Remove(consumable);
+        }
+    }
+
+    public void RemoveConsumableType(ConsumableType consumable)
+    {
+        _SuppliedTypes.Remove(consumable.ToString());
+        SuppliedTypes.Remove(consumable);
+    }
+}
+```
+
+</br>
+
+#### Migration function
+Trying to access objects whose schema is not matching any more the current one is not allowed by Realm. However, in a migration function we want to port the old data to new format, so we still need a way to access the old data. Here is when the [`Realm.DynamicApi`](https://www.mongodb.com/docs/realm-sdks/dotnet/latest/reference/Realms.Realm.html#Realms_Realm_DynamicApi) comes to the rescue *15)*; in fact this is the tool that allows us to bypass the checks for correctness that Realm normally applies. Lastly, when in a migration function the old realm, the one that needs data modification, is found in [`Migration.OldRealm`](https://docs.mongodb.com/realm-sdks/dotnet/latest/reference/Realms.Migration.html#Realms_Migration_OldRealm). With the theory out of the way, we can look at the migration function.  
```

Comment:
> It's a more general comment. I think it's a little bit confusing that the files and the line in the files are both identified by numbers. Actually maybe it could make sense to not have the numbers on the files itself? What do you think?

Mmm I don't know. I payed attention to use *n)* for in line refs while numbered links for whole code segments. If you think it's necessary I can look at using letters for whole code segments.

Reply:
Maybe LJ can chime in on the number part, I think it's a little easier if I know that one symbol is a code block and another symbol (may it be letter, number or something else) is a code line. 
Regarding anymore... I don't know 😁

Reply:
@elle-j ☝🏻😁